models/aeif_cond_beta_multisynapse.cpp

```diff
@@ -0,0 +1,701 @@
+{
+  // Initialization of the remaining members is deferred to
+  // init_buffers_().
+}
+
+aeif_cond_beta_multisynapse::Buffers_::Buffers_( const Buffers_&,
+  aeif_cond_beta_multisynapse& n )
+  : logger_( n )
+  , s_( 0 )
+  , c_( 0 )
+  , e_( 0 )
+{
+}
+
+/* ----------------------------------------------------------------
+ * Default and copy constructor for node, and destructor
+ * ---------------------------------------------------------------- */
+
+aeif_cond_beta_multisynapse::aeif_cond_beta_multisynapse()
+  : Archiving_Node()
+  , P_()
+  , S_( P_ )
+  , B_( *this )
+{
+  recordablesMap_.create();
+}
+
+aeif_cond_beta_multisynapse::aeif_cond_beta_multisynapse(
+  const aeif_cond_beta_multisynapse& n )
+  : Archiving_Node( n )
+  , P_( n.P_ )
+  , S_( n.S_ )
+  , B_( n.B_, *this )
+{
+}
+
+aeif_cond_beta_multisynapse::~aeif_cond_beta_multisynapse()
+{
+  // GSL structs may not have been allocated, so we need to protect destruction
+  if ( B_.s_ )
+  {
+    gsl_odeiv_step_free( B_.s_ );
+  }
+  if ( B_.c_ )
+  {
+    gsl_odeiv_control_free( B_.c_ );
+  }
+  if ( B_.e_ )
+  {
+    gsl_odeiv_evolve_free( B_.e_ );
+  }
+}
+
+/* ----------------------------------------------------------------
+ * Node initialization functions
+ * ---------------------------------------------------------------- */
+
+void
+aeif_cond_beta_multisynapse::init_state_( const Node& proto )
+{
+  const aeif_cond_beta_multisynapse& pr =
+    downcast< aeif_cond_beta_multisynapse >( proto );
+  S_ = pr.S_;
+}
+
+void
+aeif_cond_beta_multisynapse::init_buffers_()
+{
+  B_.spikes_.clear();   // includes resize
+  B_.currents_.clear(); // includes resize
+  Archiving_Node::clear_history();
+
+  B_.logger_.reset();
+
+  B_.step_ = Time::get_resolution().get_ms();
+
+  // We must integrate this model with high-precision to obtain decent results
+  B_.IntegrationStep_ = std::min( 0.01, B_.step_ );
+
+  if ( B_.c_ == 0 )
+  {
+    B_.c_ = gsl_odeiv_control_yp_new( P_.gsl_error_tol, P_.gsl_error_tol );
+  }
+  else
+  {
+    gsl_odeiv_control_init(
+      B_.c_, P_.gsl_error_tol, P_.gsl_error_tol, 0.0, 1.0 );
+  }
+
+  // Stepping function and evolution function are allocated in calibrate()
+
+  B_.sys_.function = aeif_cond_beta_multisynapse_dynamics;
+  B_.sys_.jacobian = NULL;
+  B_.sys_.params = reinterpret_cast< void* >( this );
+  // B_.sys_.dimension is assigned in calibrate()
+  B_.I_stim_ = 0.0;
+}
+
+void
+aeif_cond_beta_multisynapse::calibrate()
+{
+  // ensures initialization in case mm connected after Simulate
+  B_.logger_.init();
+
+  P_.receptor_types_.resize( P_.num_of_receptors_ );
+  for ( size_t i = 0; i < P_.num_of_receptors_; i++ )
+  {
+    P_.receptor_types_[ i ] = i + 1;
+  }
+
+  V_.g0_.resize( P_.num_of_receptors_ );
+
+  for ( size_t i = 0; i < P_.num_of_receptors_; ++i )
+  {
+    // denominator is computed here to check that it is != 0
+    double denom1 = P_.taus_decay[ i ] - P_.taus_rise[ i ];
+    double denom2 = 0;
+    if ( denom1 != 0 )
```

Comment:
Could you please add an informative comment why are you doing this computation and checks? Can it be already detected in the Parameters_::set?